retrieval/targetmanager.go

```diff
@@ -396,8 +396,11 @@ func (tp *prefixedTargetProvider) Run(ch chan<- *config.TargetGroup, done <-chan
 			if tg == nil {
 				break
 			}
-			tg.Source = tp.prefix(tg.Source)
-			ch <- tg
+			// Make a copy of the target group in case the discovery mechanism reuses
+			// the original object.
+			ptg := *tg
```

Comment:
At this stage the value could already be modified, no?
Wondering whether we should change the interface to only use values to begin with – or forbid the re-use in the interface contract, i.e. remove it from the Consul implementation.